sklearn/conftest.py

```diff
@@ -227,3 +229,77 @@ def pytest_configure(config):
         matplotlib.use("agg")
     except ImportError:
         pass
+
+
+# Definition of the random_seed fixture. See the docstring of the fixture for
+# more information.
+
+RANDOM_SEED_RANGE = list(range(100))  # All seeds in [0, 99] should be valid.
+random_seed_var = environ.get("SKLEARN_TESTS_RANDOM_SEED")
+if random_seed_var is None:
+    # If the environment variable is not defined, pick-up one seed at random in
+    # the range of admissible random seeds. Note, to make sure that all
+    # pytest-xdist workers see the same seed, we seed the meta-random number
+    # generator with a value derived from the year and the day.
+    rng = Random(int(datetime.now().strftime("%Y%j")))
+    random_seeds = [rng.choice(RANDOM_SEED_RANGE)]
+elif random_seed_var == "all":
+    random_seeds = RANDOM_SEED_RANGE
+else:
+    if "-" in random_seed_var:
+        start, stop = random_seed_var.split("-")
+        random_seeds = list(range(int(start.strip()), int(stop.strip()) + 1))
+    else:
+        random_seeds = [int(s.strip()) for s in random_seed_var.split(",")]
```

Comment:
Do we really want to support a list or set of seeds ? I feel like we'll only ever use a single seed or "all"

Reply:
If you run all and you get a bunch of failures, you might want just to run those seeds while iterating.

It's also useful to see how brittle a test is: I would typically iterate the test dev with `0-9` and once it's stable for that range, re-run with 'all' to confirm.

Reply:
> If you run all and you get a bunch of failures, you might want just to run those seeds while iterating.

I would do ``pytest --lf``, but I'm ok with the second argument :)